### ISSUE TITLE
There should be a warning returning Ident's with nil Identifier

### ISSUE THREAD

**Issue by alexanderkiel** [omcljs/om#749]
I killed a whole day on debugging a re-render issue only to notice that I forgot to query for my components identifier. I had the following component (greatly simplified):

```clojure
(defui MyItem
  static om/Ident
  (ident [_ {:keys [id]}]
    [:my-item/by-id id])

  static om/IQuery
  (query [_]
    [:name])
  
  Object
  (render [this]
    (let [{:keys [name]} (om/props this)]
      (dom/li nil name))))
```

I did not need the identifier in `render` and totally forgot that I need it in `ident`. That resulted in my component indexed under `[:my-item/by-id nil]` so that the reconciler couldn't find it after new data arrived from my remote.

What I suggest is that the ident check in [index-component!][1] should be extended by checking also for `nil` identifiers. I'm not sure whether it's a good idea to put this check right into `util/ident?` or only at this place.

Any suggestions?

[1]: <https://github.com/omcljs/om/blob/1.0.0-alpha41/src/main/om/next.cljs#L1153>

**Comment by anmonteiro** [comment]
It's a good use case for the `invariant` utility

**Comment by alexanderkiel** [comment]
@anmonteiro I also think so because an ident with `nil` would actually work. Do you think I should open a PR?

**Comment by anmonteiro** [comment]
fixed 4aea7fdb64067cc32c6e965aa4a8ee9d77b2620b
